internal/provider/resource/integration_circleci.go

```diff
@@ -0,0 +1,307 @@
+package resource
+
+import (
+	"context"
+	"fmt"
+	infisical "terraform-provider-infisical/internal/client"
+
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ resource.Resource = &IntegrationCircleCIResource{}
+)
+
+// NewProjectResource is a helper function to simplify the provider implementation.
+func NewIntegrationCircleCiResource() resource.Resource {
+	return &IntegrationCircleCIResource{}
+}
+
+// IntegrationCircleCI is the resource implementation.
+type IntegrationCircleCIResource struct {
+	client *infisical.Client
+}
+
+// projectResourceSourceModel describes the data source data model.
+type IntegrationCircleCIResourceModel struct {
+	CircleCIToken types.String `tfsdk:"circleci_token"`
+	ProjectID     types.String `tfsdk:"project_id"`
+
+	IntegrationAuthID types.String `tfsdk:"integration_auth_id"`
+	IntegrationID     types.String `tfsdk:"integration_id"`
+
+	Environment types.String `tfsdk:"environment"`
+	SecretPath  types.String `tfsdk:"secret_path"`
+
+	CircleCIOrgSlug   types.String `tfsdk:"circleci_org_slug"`
+	CircleCIProjectID types.String `tfsdk:"circleci_project_id"`
+}
+
+// Metadata returns the resource type name.
+func (r *IntegrationCircleCIResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_integration_circleci"
+}
+
+// Schema defines the schema for the resource.
+func (r *IntegrationCircleCIResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Create CircleCI integration & save to Infisical. Only Machine Identity authentication is supported for this data source",
+		Attributes: map[string]schema.Attribute{
+			"integration_auth_id": schema.StringAttribute{
+				Computed:      true,
+				Description:   "The ID of the integration auth, used internally by Infisical.",
+				PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+
+			"integration_id": schema.StringAttribute{
+				Computed:      true,
+				Description:   "The ID of the integration, used internally by Infisical.",
+				PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+
+			"circleci_token": schema.StringAttribute{
+				Required:      true,
+				Sensitive:     true,
+				Description:   "Your personal CircleCI token to authenticate with.",
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
```

Comment:
do we need to trigger a replace when this is updated?

Reply:
Yes, this is auth related